Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py

```diff
@@ -386,6 +368,33 @@ def PyExec(self):
             s_api.SortXAxis(InputWorkspace=probWS, OutputWorkspace=probWS, EnableLogging=False)
             self.setProperty('OutputWorkspaceProb', probWS)
 
+    def check_platform_support(self):
+        if not is_supported_f2py_platform():
+            unsupported_msg = "This algorithm can only be run on valid platforms." \
+                              + " please view the algorithm documentation to see" \
+                              + " what platforms are currently supported"
+            raise RuntimeError(unsupported_msg)
+
+    def fortran_background_encoding(self, setup_prog):
+        # fortran code uses background choices defined using the following numbers
+        setup_prog.report('Encoding input options')
+        if self._background == 'Sloping':
```

Comment:
All these if statements should be `return ['Zero','Flat','Sloping'].index(self._background)`